src/TTextEdit.cpp

```diff
@@ -1577,6 +1581,11 @@ QString TTextEdit::getSelectedText(char newlineChar)
             }
         }
         // while we are not at the end of the buffer line
+        if (mpBuffer->buffer.size() <= y) { //Check if the index will be in range first
```

Comment:
This `if` will never be reached because it is testing for the same thing as on line 1559 and if it was true on that line execution will never get to *here* as it will have already left the method.

tl;dr; This test is redundant IMHO.

Reply:
I agree. This is sadly why I stated in the discord:
Crius MacOldenlyLast Saturday at 10:43 AM
This doesn't feel right.
I can't see how else the crash could possibly happen in that function however.
and
Crius MacOldenlyLast Sunday at 7:50 AM
SlySven - Thanks. My goal is currently just to get the text selection working well enough I can delete everything and go back to coding accursed lands. This current crash bug got past all my testing, and I still don't see how it's possible, but alas, it is. Alas, testing is not something I'm good at - fixing bugs is. I got really good at fixing bugs because I write quite a few bugs. :laughing:

I honestly couldn't see any point to these tests either, but the crash bug is there, somehow. I can't figure it out. :(

Thank you, you have confirmed what I was feeling about these checks. There is still a crash bug in there somewhere, and I just can't see where. :(